plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/AccountSelector.java

```diff
@@ -130,6 +140,20 @@ public int getAccountCount() {
     return combo.getItemCount() - 1;  // <Add a new account...> is always in the combo
   }
 
+  private void fireSelectionListeners() {
+    for (Object o : selectionListeners.getListeners()) {
+      ((Runnable) o).run();
```

Comment:
You're not executing these in separate threads, so maybe Callable instead of Runnable?